x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -0,0 +1,310 @@
+        RealmSettings.realmSettingPrefix(TYPE),
+        "populate_user_metadata",
+        key -> Setting.boolSetting(key, DEFAULT_POPULATE_USER_METADATA, Setting.Property.NodeScope)
+    );
+
+    // Client TLS settings for incoming connections (subset of SSLConfigurationSettings)
+
+    public static final Setting.AffixSetting<String> CLIENT_AUTHENTICATION_TYPE = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "client_authentication.type",
+        key -> Setting.simpleString(key, DEFAULT_CLIENT_AUTHENTICATION_TYPE, value -> {
+            if (SUPPORTED_CLIENT_AUTHENTICATION_TYPE.contains(value) == false) {
+                throw new IllegalArgumentException(
+                    "Invalid value [" + value + "] for [" + key + "]. Allowed values are " + SUPPORTED_CLIENT_AUTHENTICATION_TYPE + "}]"
+                );
+            }
+        }, Setting.Property.NodeScope)
+    );
+    public static final Setting.AffixSetting<Optional<String>> CLIENT_AUTHENTICATION_TRUSTSTORE_PATH =
+        SSLConfigurationSettings.TRUSTSTORE_PATH.affixSetting(CLIENT_AUTHENTICATION_PREFIX, "");
+    public static final Setting.AffixSetting<SecureString> CLIENT_AUTHENTICATION_TRUSTSTORE_PASSWORD =
+        SSLConfigurationSettings.TRUSTSTORE_PASSWORD.affixSetting(CLIENT_AUTHENTICATION_PREFIX, "");
+    public static final Setting.AffixSetting<SecureString> CLIENT_AUTHENTICATION_LEGACY_TRUSTSTORE_PASSWORD =
+        SSLConfigurationSettings.LEGACY_TRUSTSTORE_PASSWORD.affixSetting(CLIENT_AUTHENTICATION_PREFIX, "");
+    public static final Setting.AffixSetting<String> CLIENT_AUTHENTICATION_TRUSTSTORE_ALGORITHM =
+        SSLConfigurationSettings.TRUSTSTORE_ALGORITHM.affixSetting(CLIENT_AUTHENTICATION_PREFIX, "");
+    public static final Setting.AffixSetting<Optional<String>> CLIENT_AUTHENTICATION_TRUSTSTORE_TYPE =
+        SSLConfigurationSettings.TRUSTSTORE_TYPE.affixSetting(CLIENT_AUTHENTICATION_PREFIX, "");
+    public static final Setting.AffixSetting<List<String>> CLIENT_AUTHENTICATION_CERT_AUTH_PATH = SSLConfigurationSettings.CERT_AUTH_PATH
+        .affixSetting(CLIENT_AUTHENTICATION_PREFIX, "");
+    // Note: <allowed_issuer>.client_authentication_shared_secret not defined here. It goes in the Elasticsearch keystore setting.
+
+    // Individual Cache settings
+
+    public static final Setting.AffixSetting<TimeValue> CACHE_TTL = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "cache.ttl",
+        key -> Setting.timeSetting(key, DEFAULT_CACHE_TTL, Setting.Property.NodeScope)
+    );
+
+    public static final Setting.AffixSetting<Integer> CACHE_MAX_USERS = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "cache.max_users",
+        key -> Setting.intSetting(key, DEFAULT_CACHE_MAX_USERS, MIN_CACHE_MAX_USERS, Setting.Property.NodeScope)
+    );
+
+    // Individual outgoing HTTP settings
```

Comment:
These settings are shared with PkiRealm. Later, it might be nice to put them in a shared class and reuse it in both realms.